torch/_refs/__init__.py

```diff
@@ -467,21 +476,38 @@ def _isfinite(a: TensorLikeType) -> TensorLikeType:
 
 
 def _isinf(a: TensorLikeType) -> TensorLikeType:
-    # TODO Add complex tensor support to remove is_infinite prim
-    # if utils.is_complex_dtype(a):
-    #     return bitwise_or(_isinf(real(a), _isinf(imag(a))
-    # else:
-    #     return bitwise_not(bitwise_or(isnan(a), isfinite(a)))
-    if utils.is_float_dtype(a.dtype) or utils.is_complex_dtype(a.dtype):
-        return prims.is_infinite(a)
-
-    return zeros_like(a, dtype=torch.bool)
+    if utils.is_complex_dtype(a.dtype):
+        return logical_or(_isinf(real(a)), _isinf(imag(a)))
+    # TODO: use logical_not instead of bitwise_not
+    return logical_and(bitwise_not(isfinite(a)), bitwise_not(_isnan(a)))
 
 
 isinf = _make_elementwise_unary_reference(
     _isinf,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.ALWAYS_BOOL,
-    aten_op=torch.ops.aten.isinf,  # prim/aten name mismatch
+    aten_op=torch.ops.aten.isinf,
+)
+
+
+def _isposinf(a: TensorLikeType) -> TensorLikeType:
+    return logical_and(_isinf(a), gt(a, 0))
+
+
+isposinf = _make_elementwise_unary_reference(
+    _isposinf,
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.ALWAYS_BOOL,
+    aten_op=torch.ops.aten.isposinf,
+)
+
+
+def _isneginf(a: TensorLikeType) -> TensorLikeType:
+    return logical_and(_isinf(a), lt(a, 0))
```

Comment:
```
return eq(a, float('-inf'))
```

?

Reply:
See https://github.com/pytorch/pytorch/pull/78655/files#r889423851

Reply:
You're right, sorry, but I think we can just add a short-circuit for integer (and boolean) tensors

Reply:
check for complex and short-circuit for integers added.

Reply:
Looks like `ref_executor` does not support that. Looking deep into it.